src/main/java/com/github/packageurl/PackageURL.java

```diff
@@ -464,13 +465,56 @@ private static boolean isUnreserved(int c) {
     }
 
     private static boolean isAlpha(int c) {
-        return ((c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z'));
+        return (isLowerCase(c) || isUpperCase(c));
     }
 
     private static boolean isDigit(int c) {
         return (c >= '0' && c <= '9');
     }
 
+    private static boolean isUpperCase(int c) {
+        return (c >= 'A' && c <= 'Z');
+    }
+
+    private static int indexOfFirstUpperCaseChar(String s) {
+        int length = s.length();
+
+        for (int i = 0; i < length; i++) {
+            if (!isLowerCase(s.charAt(i))) {
```

Comment:
Improve readability:
```suggestion
            if (isUpperCase(s.charAt(i))) {
```